### PR TITLE
Mongo-Optimization-task.margo.astreyko

### PR DESCRIPTION
**Pull request by MargaritaAstreiko**
https://travis-ci.org/github/Honey-badger1/db-assignments

### CODE REVIEW

**Comment by KostyaKov** [comment]
Can you add travis link with task_3_1, please.

I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

**Comment by MargaritaAstreiko** [comment]
https://travis-ci.org/github/Honey-badger1/db-assignments/builds/715284785

File: task/mongo-optimization-task.js [hunk 1]
```diff
@@ -209,97 +251,131 @@ async function task_3_1(db) {
...
+                            value: 1,
+                            "versions.definition": 1,
+                            label: 1,
+                            definition: 1
+                        }
+                    },
+                    {
+                        $match: {
+                            $expr: {
+                                $and:
```
Comment (IlyaMokin) [inline task/mongo-optimization-task.js]: - why you need `$and` here?
- can you avoid the second `$match`?
- try to add `$limit` at the end of the pipeline

File: task/mongo-tasks.js [hunk 2]
```diff
@@ -308,4 +308,4 @@ module.exports = {
     task_1_20: task_1_20,
     task_1_21: task_1_21,
     task_1_22: task_1_22
-};
+};
```
Comment (IlyaMokin) [inline task/mongo-tasks.js]: Redundant change for the task

**Comment by IlyaMokin** [comment]
looking good! Thanks for the efforts. The PR is accepted

**Pull request closed without merging**